phlat/run.b38.sh

```diff
@@ -0,0 +1,164 @@
+  --b2url         the path to where bowtie2 is located, DEFAULT /usr/bin/bowtie2 
+  --fastq1        the name of seq num 1, DEFAULT 'example_1.fastq.gz' 
+  --fastq2        the name of seq num 2, DEFAULT 'example_2.fastq.gz'
+  --ref-fasta     the path to where the reference fasta file is located, NO DEFAULT
+
+
+EOF
+}
+
+# die and opts based on this example
+# http://mywiki.wooledge.org/BashFAQ/035
+# --long-opt* example here
+# https://stackoverflow.com/a/7069755
+function die {
+    printf '%s\n' "$1" >&2 && show_help && exit 1
+}
+
+# check arguments
+while test $# -gt 0; do
+    case $1 in
+        -h|--help)
+            show_help
+            exit
+            ;;
+         --phlat-dir*)
+            if [ ! "$2" ]; then
+                PHLAT_DIR=""
+            else
+                PHLAT_DIR=$2
+                shift
+            fi
+            ;;
+         --data-dir*)
+	    if [ ! "$2" ]; then
+		DARA_DIR=""
+	    else
+		DATA_DIR=$2
+		shift
+	    fi
+	    ;;
+         --samtools*)
+	    if [ ! "$2" ]; then
+		SAMTOOLS=""
+	    else
+		SAMTOOLS=$2
+		shift
+	    fi
+	    ;;
+         --bam*)
+            if [ ! "$2" ]; then 
+                die 'ERROR: "--bam" requires a non-empty argument.'
+            else
+                BAM=$2
+                shift
+            fi
+            ;;
+        --index-dir*)
+	    if [ ! "$2" ]; then
+		INDEX_DIR=""
+	    else
+		INDEX_DIR=$2
+		shift
+	    fi
+       	    ;;
+	--rs-dir*)
+	    if [ ! "$2" ]; then
+		RS_DIR=""
+	    else
+		RS_DIR=$2
+		shift
+	    fi
+       	    ;;
+	--b2url*)
+	    if [ ! "$2" ]; then
+		B2URL=""
+	    else
+		B2URL=$2
+		shift
+	    fi
+       	    ;;
+        --fastq1*)
+	    if [ ! "$2" ]; then
+		FASTQ1=""
+	    else
+		FASTQ1=$2
+		shift
+	    fi
+       	    ;;
+       --fastq2*)
+	    if [ ! "$2" ]; then
+		FASTQ2=""
+	    else
+		FASTQ2=$2
+		shift
+	    fi
+       	    ;;
+        --ref-fasta*)
+	    if [ ! "$2" ]; then
+		REF_FASTA="ERROR: --ref-fasta requires non-empty argument"
+	    else
+		REF_FASTA=$2
+		shift
+	    fi
+       	    ;; 
+
+        *)
+            break
+            ;;
+    esac
+    shift
+done
+
+# double check all vars are set up
+[ -z $PHLAT_DIR    ] && PHLAT_DIR="$SRC_DIR/phlat-release"
+[ -z $DATA_DIR     ] && DATA_DIR="example"
+[ -z $SAMTOOLS     ] && SAMTOOLS="/usr/local/bin/samtools"
+[ -z $BAM          ] && die "Missing argument --bam"
+[ -z $INDEX_DIR    ] && INDEX_DIR="$PHLAT_DIR/b2folder"
+[ -z $RS_DIR       ] && RS_DIR="$DATA_DIR/results" 
+[ -z $B2URL        ] && B2URL="/usr/bin/bowtie2"
+[ -z $FASTQ1       ] && FASTQ1="example_1.fastq.gz"
+[ -z $FASTQ2       ] && FASTQ2="example_2.fastq.gz"
+[ -z $REF_FASTA    ] && die "Missing argument --ref-fasta"
+
+
+mkdir -p $DATA_DIR/results
+tmpdir=$DATA_DIR/tmp
+mkdir -p $tmpdir
+
+# extract hla regions and unmapped reads
+echo "extracting hla region and unmapped reads ..."
+$SAMTOOLS view -h -T $REF_FASTA $BAM chr6:29836259-33148325 >$tmpdir/reads.sam
```

Comment:
In the context of the larger pipelines, (as opposed to a one-off bash script it came from) I don't like that this has a hardcoded human build38 path. (if we tried to run this on b37 data (or mouse data), it would happily do the wrong thing!).  The ideal thing to do here would probably be to use the annotation build info to look up the HLA locus, I think?  That could look something like this:
1) grep the GTF file for the proper gene names (`HLA-A, HLA-B, HLA-C,`, etc) to get coordinates
2) use those coordinates in this command